src/Symfony/Component/Form/ChoiceList/Loader/CallbackChoiceLoader.php

```diff
@@ -0,0 +1,79 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Form\ChoiceList\Loader;
+
+use Symfony\Component\Form\ChoiceList\ArrayChoiceList;
+
+/**
+ * Loads a {@link ArrayChoiceList} interface from any callable.
+ *
+ * @author Jules Pietri <jules@heahprod.com>
+ */
+class CallbackChoiceLoader implements ChoiceLoaderInterface
+{
+    /**
+     * The callable used to load the choices.
+     *
+     * @var callable
+     */
+    private $callback;
+
+    /**
+     * The loaded choice list.
+     *
+     * @var ArrayChoiceList
+     */
+    private $choiceList;
+
```

Comment:
Sorry, I was not very clear. Let me try again. It would be better to add a phpdoc to the constructor:

```php
 /**
  * @param callable $callback The callable used to load the array of choices
  */
public function __construct(callable $callback)
 {
     $this->callback = $callback;
}
```

and remove the one on the private property:

```php
private $callback;
```